src/promptflow/tests/sdk_cli_test/recording_utilities/record_storage.py

```diff
@@ -159,7 +158,88 @@ def _recursive_create_hashable_args(self, item):
         else:
             return item
 
-    def set_record(self, input_dict: Dict, output) -> None:
+    def _parse_output_generator(self, output):
+        output_type = ""
+        output_value = None
+        output_generator = None
+        if isinstance(output, dict):
+            output_value = {}
+            output_generator = {}
+            for item in output.items():
+                k, v = item
+                if type(v).__name__ == "generator":
+                    vlist = list(v)
+
+                    def vgenerator():
+                        for vitem in vlist:
+                            yield vitem
+
+                    output_value[k] = vlist
+                    output_generator[k] = vgenerator()
+                    output_type = "dict[generator]"
+                else:
+                    output_value[k] = v
+        elif isinstance(output, list):
+            output_value = []
+            output_generator = []
+            for item in output:
+                if type(item).__name__ == "generator":
+                    ilist = list(item)
+
+                    def igenerator():
+                        for i in ilist:
+                            yield i
+
+                    output_value.append(ilist)
+                    output_generator.append(igenerator())
+                    output_type = "list[generator]"
+                else:
+                    output_value.append(item)
+        elif type(output).__name__ == "generator":
+            output_value = list(output)
+
+            def generator():
+                for item in output_value:
+                    yield item
+
+            output_generator = generator()
+            output_type = "generator"
+        else:
+            output_value = output
+            output_generator = None
+            output_type = type(output).__name__
+        return output_value, output_generator, output_type
+
+    def _create_output_generator(self, output, output_type):
+        output_generator = None
+        if output_type == "dict[generator]":
```

Comment:
Could you share some example on these two scenarios: dict[generator] dict[generator]?

Reply:
for list[generator]: There is no examples, but I'm not sure.